R/repeat_linter.R

```diff
@@ -0,0 +1,46 @@
+#' Repeat linter
+#'
+#' Check that `while` is not used for infinite loops.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "while (TRUE) { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' # okay
+#' lint(
+#'   text = "repeat { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' @evalRd rd_tags("repeat_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+repeat_linter <- function() {
+  xpath <- "
+  //WHILE[following-sibling::*[1][self::OP-LEFT-PAREN] and
+  following-sibling::*[2][self::expr[NUM_CONST[text()='TRUE'] and count(*)=1]]
+  and following-sibling::*[3][self::OP-RIGHT-PAREN]]"
```

Comment:
this is also redundant. if you're worried about more complicated expressions in the condition, your test for NUM_CONST is implicitly handling that -- TRUE would show up at further nesting of <expr> in that case.

Reply:
I implemented it as suggested but had to test that an expression existed at all